vignettes/modification.Rmd

```diff
@@ -119,6 +126,42 @@ xml_add_child(t1, read_xml("<test/>"))
 x
 ```
 
+## Removing nodes ##
+The `xml_remove()` can be used to remove a node (and it's children) from a
+tree. The default behavior is to unlink the node from the tree, but does _not_
+free the memory for the node, so R objects pointing to the node are still
+valid.
+
+This allows code like the following to work without crashing R
+
+```r
+x <- read_xml("<foo><bar><baz/><bar></foo>")
+x1 <- x %>% xml_children() %>% .[[1]]
+x2 <- x1 %>% xml_children() %>% .[[1]]
+
+xml_remove(x1)
+rm(x1)
+gc()
+
+x2
+```
+If you are not planning on referencing these nodes again this memory is wasted.
+Calling `xml_remove(free = TRUE)` will remove the nodes _and_ free the memory
+used to store them.  **Note** In this case _any_ node which previously pointed
+to the node or it's children will instead be pointing to free memory and may
+cause R to crash. It is the responsibility of the user to remove any objects
```

Comment:
I'd make this a little friendlier: "xml2 can't figure this out for you, so it's your responsibility to remove..."